python/dgl/data/dgl_dataset.py

```diff
@@ -0,0 +1,157 @@
+"""Basic DGL Dataset
+"""
+
+from __future__ import absolute_import
+
+import os, sys
+from .utils import download, extract_archive, get_download_dir, makedirs
+from ..utils import retry_method_with_fix
+
+class DGLDataset(object):
```

Comment:
Will we introduce the `F.dataset` base class now or in the later version?

Reply:
Maybe later